pyface/font.py

```diff
@@ -0,0 +1,313 @@
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Toolkit-independent font utilities.
+
+Pyface fonts are intended to be generic, but able to be mapped fairly well
+to most backend toolkit font descriptions.  In most cases we can describe
+fonts along the common dimensions that are used by CSS, Wx, and Qt.  However
+when it comes to actually working with a font, the toolkit needs to take the
+description and produce something that is as close as possible to the
+specification, but within the constraints of the toolkit, operating system
+and available fonts on the machine where this is being executed.
+
+Font Properties
+---------------
+
+The properties that fonts have are:
+
+Font Family
+    A list of font family names in order of preference, such as "Helvetica"
+    or "Comic Sans".  There are several generic font family names that can
+    be used as fall-backs in case all preferred fonts are unavailable.  In
+    the case of a font that has been selected by the toolkit this list will
+    have one value which is the actual font family name.
+
+Weight
+    How thick or dark the font glyphs are.  These can be given as a number
+    from 1 (lightest) to 999 (darkest), but are typically specified by a
+    multiple of 100 from 100 to 900, with a number of synonyms such as 'light'
+    and 'bold' available for those values.
+
+Stretch
+    The amount of horizontal compression or expansion to apply to the glyphs.
+    These can be given as a percentage between 50% and 200%, or by strings
+    such as as 'condensed' and 'expanded' that correspond to those values.
+
+Style
+    This selects either 'oblique' or 'italic' variants typefaces of the given
+    font family.  If neither is wanted, the value is 'normal'.
+
+Size
+    The overall size of the glyphs. This can be expressed either as the
+    numeric size in points, or as a string such as "small" or "large".
+
+Variants
+    A set of additional font style specifiers, such as "small-caps",
+    "strikethrough", "underline" or "overline", where supported by the
+    underlying toolkit.
+
+Font Specificiation Class
+-------------------------
+
+The Pyface Font class is a HasStrictTraits class which specifies a requested
+font.  It has traits for all of the font properties, plus additional utility
+methods that produce modifed versions of the font.
+
+It also has methods that convert the Font class to and from a toolkit Font
+class.
+
+"""
+import re
+
+from traits.api import (
+    BaseCFloat, CList, CSet, DefaultValue, Enum, HasStrictTraits, Map, Range,
+    Str, TraitError, TraitType
+)
+from traits.trait_type import NoDefaultSpecified
+
+WEIGHTS = {str(i): i for i in range(100, 1001, 100)}
+
+# Note: we don't support 'medium' as an alias for weight 500 because it
+# conflicts with the usage of 'medium' as an alias for a 12pt font in the CSS
+# specification for font attributes.
+WEIGHTS.update({
```

Comment:
These come from the CSS spec, Qt and Wx constant names, true/open type font conventions, and in some cases (originally) from "traditional" values from the typesetting world (eg. the font weights date back to the fonts used in phototypesetting systems).

The CSS spec is probably the simplest comprehensive reference.

Reply:
Looks like we have some extra weights here compared to the CSS spec: https://www.w3.org/TR/css-fonts-3/#propdef-font-weight

Reply:
More specifically, these are the values from CSS: `normal | bold | bolder | lighter | 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900`
Where do the other values come from?

Reply:
As mentioned above, this wasn't just from the CSS spec - in particular it should at least include the mappings that Enable has here https://github.com/enthought/enable/blob/db84efa4d031fd641ebe097301a56764634ce21d/kiva/fonttools/_constants.py#L90-L105
as well as the names that Qt uses for weight constants: https://doc.qt.io/qt-5/qfont.html#Weight-enum
and the names wx uses: https://wxpython.org/Phoenix/docs/html/wx.FontWeight.enumeration.html
and the OpenType font weights: https://docs.microsoft.com/en-us/typography/opentype/spec/os2#usweightclass

More info can be found here: https://bigelowandholmes.typepad.com/bigelow-holmes/2015/07/on-font-weight.html

This is all very fuzzy because the names predate computerized fonts and most font families don't supply all weights.